src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -281,7 +281,21 @@ void MemoryInput::initSupportedPrimitiveDescriptors() {
 
 void MemoryInput::initOptimalPrimitiveDescriptor() {
     // Mimic the child node memory desc to avoid extra reorder
-    auto childEdge = getChildEdgeAt(0);
+    auto selectChild = [&]() -> EdgePtr {
+        for (auto&& edge : getChildEdgesAtPort(0)) { // always only one child port
+            // prefer SDPA
+            auto child = edge->getChild();
+            if (one_of(child->getType(), Type::ScaledDotProductAttention)) {
+                return edge;
+            }
+        }
+
+        // default?
+        auto childEdge = getChildEdgeAt(0);
+        return childEdge;
+    };
+
+    auto childEdge = selectChild();
```

Comment:
This code is not relevant anymore since we are preparing some significant changes to the entire states subsystem.